components/autocomplete/index.js

```diff
@@ -330,11 +338,25 @@ export class Autocomplete extends Component {
 		// create a regular expression to filter the options
 		const search = open ? new RegExp( '(?:\\b|\\s|^)' + escapeRegExp( query ), 'i' ) : /./;
 		// filter the options we already have
-		const filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
+		let filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
 		// check if we should still suppress the popover
 		const suppress = ( open && wasSuppress === open.idx ) ? wasSuppress : undefined;
 		// update the state
 		if ( wasOpen || open ) {
+			if ( open ) {
+				if ( this.props.completers[ open.idx ].setSearch ) {
```

Comment:
To make this clearer to read in the multiple places it is used I would extract `this.props.completers[ open.idx ]` into a constant, ie:
```javascript
const completer = this.props.completers[ open.idx ];
```